### PR TITLE
[QF-3598] Fix voice search continuing after selecting a result

### PR DESCRIPTION
**Pull request by zonetecde**
## Summary

This PR fixes an issue related to the **voice search** feature.

Previously, if a user was speaking and the suggestion **“Search for [spoken text]”** appeared, clicking on it *before* the browser detected the end of speech would:
- Redirect the user to `/search?query=[spoken text]`
- **But** keep the search modal open
- And continue capturing microphone input, appending more text unintentionally

To fix this, the `SpeechRecognitionInterface` was made global (effectively a singleton).  
This allows `SearchResultItem/index.tsx` to access it and **stop the browser microphone immediately** when a search result is clicked.

Closes: [QF-3598](https://quranfoundation.atlassian.net/browse/QF-3598)

## Type of Change

- [X] 🐛 Bug fix (non-breaking change)

## Scope Confirmation

- [X] This PR addresses **one** fix only  
- [ ] If multiple changes were needed, they are split into separate PRs

## Rollback Safety

- [X] Can be safely reverted without data issues or migrations

## Test Plan

- [ ] Unit tests added/updated  
- [ ] Integration tests added/updated  
- [X] Manual testing performed  

Because this change involves microphone access, testing was done manually.

### Testing steps

1. Go to `/search`  
2. Click the microphone to start a voice search  
3. Keep speaking until **“Search for …”** appears, without stopping  
4. Quickly click on **“Search for …”** while still speaking  
5. Verify that:
   - The user is redirected to `/search?query=x`
   - The microphone stops immediately
   - The search results modal is closed

### Edge Cases Verified

- [X] ⏳ Loading state handled  
- [ ] ❌ Error state handled  
- [ ] 📭 Empty state handled  
- [ ] 👤 Logged-in vs guest behavior  

## Code Quality

- [X] Self-review completed (file by file)  
- [X] Follows project style guidelines  
- [X] No `any` types used (or justified if unavoidable)  
- [X] No unused code, imports, or dead code  
- [X] Inline comments explain complex logic (“why”)  
- [X] Functions under 30 lines and single-responsibility  

## Testing & Validation

- [X] All tests pass locally (`yarn test`)  
- [X] Linting passes (`yarn lint`)  
- [X] Build succeeds (`yarn build`)  
- [X] Edge cases and error scenarios handled  

## Documentation

- [X] Code is self-documenting with clear naming  
- [ ] README updated (if adding features or setup changes)  
- [X] Inline comments added for complex logic  

## Screenshots / Videos

https://github.com/user-attachments/assets/83bfa38e-ae1c-455a-95b6-f06cf5af40e7

## AI Assistance Disclosure

- [ ] AI tools were NOT used for this PR  
- [X] AI tools were used, and I have **thoroughly reviewed and validated** all generated code

[QF-3598]: https://quranfoundation.atlassian.net/browse/QF-3598?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/hooks/useVoiceSearch.ts [hunk 1]
```diff
@@ -53,6 +55,22 @@ const useVoiceSearch = (options: UseVoiceSearchOptions) => {
...
 
+  // Effect to stop speech recognition when microphone is deactivated
+  useEffect(() => {
+    if (!isMicActive) {
+      if (sharedSpeechRecognitionInstance) {
+        try {
+          sharedSpeechRecognitionInstance.stop();
+        } catch (error) {
+          // eslint-disable-next-line no-console
+          console.error('Error stopping speech recognition', error);
```
Comment (osamasayed) [inline src/hooks/useVoiceSearch.ts]: No console logging. We should send it to Sentry instead

File: src/hooks/useVoiceSearch.ts [hunk 2]
```diff
@@ -142,10 +163,17 @@ const useVoiceSearch = (options: UseVoiceSearchOptions) => {
...
   const initializeSpeechRecognition = useCallback(() => {
-    if (speechRecRef.current) return true;
-
     try {
+      if (sharedSpeechRecognitionInstance) {
+        try {
+          sharedSpeechRecognitionInstance.stop();
+        } catch (error) {
+          // eslint-disable-next-line no-console
+          console.error('Error stopping speech recognition', error);
```
Comment (osamasayed) [inline src/hooks/useVoiceSearch.ts]: Same
